admin/category_product_listing.php

```diff
@@ -693,13 +758,25 @@
             }
 
             $products_count = 0;
+
+            // -----
+            // Give a watching observer the chance to modify the products' query to gather additional fields for the display.
+            //
+            // Notes:
+            // 1. Any modification of the $extra_select must include a leading comma (,).
+            // 2. Any modification of the $extra_from must include a leading comma (,).
+            // 3. Any modification of the $extra_ands must include a leading ' AND' ... note the leading space requirement!
+            //
+            $extra_select = $extra_from = $extra_joins = $extra_ands = '';
+            $zco_notifier->notify('NOTIFY_ADMIN_PROD_LISTING_PRODUCTS_QUERY', '', $extra_select, $extra_from, $extra_joins, $extra_ands, $order_by);
+            
             $products_query_raw = "SELECT p.products_type, p.products_id, pd.products_name, p.products_quantity,
                                           p.products_price, p.products_status, p.products_model, p.products_sort_order,
-                                          p.master_categories_id
-                                   FROM " . TABLE_PRODUCTS . " p
+                                          p.master_categories_id $extra_select
+                                   FROM " . TABLE_PRODUCTS . " p $extra_from
                                    LEFT JOIN " . TABLE_PRODUCTS_DESCRIPTION . " pd ON (pd.products_id = p.products_id)";
-
-            $where = " WHERE pd.language_id = " . (int)$_SESSION['languages_id'];
+            $products_query_raw .= $extra_joins;
+            $where = " WHERE pd.language_id = " . (int)$_SESSION['languages_id'] .$extra_ands;
```

Comment:
Formatting request:

Please always leave one space on either side of a concatenating `.` operator.